src/main/java/com/maddyhome/idea/vim/extension/searchindex/VimSearchIndexExtension.kt

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2003-2025 The IdeaVim authors
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/MIT.
+ */
+package com.maddyhome.idea.vim.extension.searchindex
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.ProjectManager
+import com.intellij.openapi.wm.StatusBar
+import com.intellij.openapi.wm.StatusBarWidget
+import com.intellij.openapi.wm.StatusBarWidgetFactory
+import com.intellij.openapi.wm.WindowManager
+import com.intellij.openapi.wm.impl.status.EditorBasedWidget
+import com.intellij.util.Consumer
+import com.maddyhome.idea.vim.VimPlugin
+import com.maddyhome.idea.vim.api.ExecutionContext
+import com.maddyhome.idea.vim.api.VimEditor
+import com.maddyhome.idea.vim.api.injector
+import com.maddyhome.idea.vim.api.options
+import com.maddyhome.idea.vim.command.MappingMode
+import com.maddyhome.idea.vim.command.OperatorArguments
+import com.maddyhome.idea.vim.common.ModeChangeListener
+import com.maddyhome.idea.vim.common.TextRange
+import com.maddyhome.idea.vim.extension.ExtensionHandler
+import com.maddyhome.idea.vim.extension.VimExtension
+import com.maddyhome.idea.vim.extension.VimExtensionFacade
+import com.maddyhome.idea.vim.newapi.IjVimSearchGroup
+import com.maddyhome.idea.vim.newapi.ij
+import com.maddyhome.idea.vim.newapi.vim
+import com.maddyhome.idea.vim.state.mode.Mode
+import com.maddyhome.idea.vim.ui.ex.ExEntryPanel
+import java.awt.Component
+import java.awt.event.MouseEvent
+import javax.swing.KeyStroke
+
+internal object SearchIndex {
+  internal const val ID = "IdeaVimSearchIndex"
+  internal const val DISPLAY_NAME = "IdeaVim searchindex"
+  private var currentText = ""
+
+  fun update(text: String) {
+    currentText = text
+    val windowManager = WindowManager.getInstance()
+    ProjectManager.getInstance().openProjects.forEach {
+      val statusBar = windowManager.getStatusBar(it)
+      statusBar.updateWidget(ID)
+    }
+  }
+
+  fun getCurrentText(): String = currentText
+}
+
+internal class SearchIndexWidgetFactory : StatusBarWidgetFactory {
+  override fun getId() = SearchIndex.ID
+  override fun getDisplayName() = SearchIndex.DISPLAY_NAME
+  override fun createWidget(project: Project) = SearchIndexWidget(project)
+  override fun disposeWidget(widget: StatusBarWidget) {}
+  override fun canBeEnabledOn(statusBar: StatusBar) = true
+  override fun isAvailable(project: Project) = true
```

Comment:
Since this is an extension, I'd say that this widget should be available only in case the extension is enabled.